fdbserver/tester.actor.cpp

```diff
@@ -852,7 +852,14 @@ ACTOR Future<Void> clearData(Database cx) {
 			break;
 		} catch (Error& e) {
 			TraceEvent(SevWarn, "TesterClearingDatabaseError", debugID).error(e);
+			state int err_code = e.code();
 			wait(tr.onError(e));
+			if (err_code == error_code_transaction_too_old) {
+				// If this transaction is conflicted, the effect is still in the
```

Comment:
That's not the error code you get for a transaction conflict. Even so though, a previous attempt at committing a self-conflicting transaction does not affect subsequent retries, since the previous attempt is no longer in-flight. (Unless causal write risky is set)

Reply:
You are right. The problem is the buggified knob `MAX_WRITE_TRANSACTION_LIFE_VERSIONS` can be only 1M. In some tests, this transaction always end up `commitVersion - readVersion` is a little above 1M, thus always getting `transaction_too_old` here. I have a workaround by setting the knob to its default value. Maybe you have better ideas?